tests/embed/test_n2v.py

```diff
@@ -1,25 +1,225 @@
+        vocab_list = model[1]
+        self.assertIsNotNone(model)
+        self.assertIsNotNone(model[0])
+        self.assertIsNotNone(model[1])
+
+        # model matrix should be 34 x 128
+        self.assertEqual(model_matrix.shape[0], 34)
+        self.assertEqual(model_matrix.shape[1], 128)
+
+        # vocab list should have exactly 34 elements
+        self.assertEqual(len(vocab_list), 34)
+
+    def test_node2vec_embedding_florentine_graph_correct_shape_is_returned(self):
+        graph = nx.florentine_families_graph()
+        for s, t in graph.edges():
+            graph.add_edge(s, t, weight=1)
+
+        model = gc.embed.node2vec_embed(graph)
+        model_matrix: np.ndarray = model[0]
+        vocab_list = model[1]
+        self.assertIsNotNone(model)
+        self.assertIsNotNone(model[0])
+        self.assertIsNotNone(model[1])
+
+        # model matrix should be 34 x 128
+        self.assertEqual(model_matrix.shape[0], 15)
+        self.assertEqual(model_matrix.shape[1], 128)
+
+        # vocab list should have exactly 34 elements
+        self.assertEqual(len(vocab_list), 15)
+
+    def test_node2vec_embedding_barbell_graph_correct_shape_is_returned(self):
+        graph = nx.barbell_graph(25, 2)
+        for s, t in graph.edges():
+            graph.add_edge(s, t, weight=1)
+
+        model = gc.embed.node2vec_embed(graph)
+        model_matrix: np.ndarray = model[0]
+        vocab_list = model[1]
+        self.assertIsNotNone(model)
+        self.assertIsNotNone(model[0])
+        self.assertIsNotNone(model[1])
+
+        # model matrix should be 34 x 128
+        self.assertEqual(model_matrix.shape[0], 52)
+        self.assertEqual(model_matrix.shape[1], 128)
+
+        # vocab list should have exactly 34 elements
+        self.assertEqual(len(vocab_list), 52)
+
+    def test_get_walk_length_lower_defaults_to_1(self):
+        expected_walk_length = 1
+
+        g = gc.embed.n2v._Node2VecGraph(nx.Graph(), 1, 1)
+        w = g._get_walk_length_interpolated(
+            degree=0, percentiles=[1, 2, 3, 4, 10, 100], max_walk_length=10
+        )
+
+        self.assertEqual(w, expected_walk_length)
+
+    def test_get_walk_length_higher_default_to_walk_length(self):
+        expected_walk_length = 100
+
+        g = gc.embed.n2v._Node2VecGraph(nx.Graph(), 1, 1)
+        w = g._get_walk_length_interpolated(
+            degree=10,
+            percentiles=[2, 3, 4, 5, 6, 7, 8, 9],
+            max_walk_length=expected_walk_length,
+        )
+
+        self.assertEqual(w, expected_walk_length)
+
+    def test_get_walk_length_in_middle_selects_interpolated_bucket(self):
+        expected_walk_length = 5
+
+        g = gc.embed.n2v._Node2VecGraph(nx.Graph(), 1, 1)
+        w = g._get_walk_length_interpolated(
+            degree=5, percentiles=[2, 3, 4, 5, 6, 7, 8, 9], max_walk_length=10
+        )
+
+        self.assertEqual(w, expected_walk_length)
+
 
-if __name__ == "__main__":
-    unittest.main()
+_edge_list = """
```

Comment:
we have a test utils and test_data package and you could stash this in there if you wanted.  I'd prefer that over adding it inline like this

Reply:
I can move it - this change is a result from the merging the other test file, not an added code change for this pr.